src/serialization/SerializationService.ts

```diff
@@ -364,29 +390,29 @@ export class SerializationServiceV1 implements SerializationService {
         if (candidate == null) {
             return;
         }
-        this.registerSerializer('!global', candidate);
+        this.registerSerializer(SerializationSymbols.GLOBAL_SYMBOL, candidate);
     }
 
     private static isCustomSerializable(object: any): boolean {
         const prop = 'hzCustomId';
         return (typeof object[prop] === 'number' && object[prop] >= 1);
     }
 
-    private findSerializerByName(name: string, isArray: boolean): Serializer {
-        let convertedName: string;
-        if (name === 'number') {
-            convertedName = this.serializationConfig.defaultNumberType;
-        } else if (name === 'buffer') {
-            convertedName = 'byteArray';
-        } else {
-            convertedName = name;
-        }
-        const serializerName = convertedName + (isArray ? 'Array' : '');
-        const serializerId = this.serializerNameToId[serializerName];
-        if (serializerId == null) {
-            return null;
+    // eslint-disable-next-line @typescript-eslint/ban-types    
+    private findSerializerByName(clazz: Function | Symbol, isArray: boolean): Serializer {
+        const clazzType = typeof clazz == 'symbol' ? clazz : clazz.constructor;
+        const isArrayChange = (clazz.constructor == Buffer) ? true : isArray;
+        const objectKeyValue = getTypes(clazzType);
```

Comment:
What about changing method name to getKeyForClass? We can make the method accept only `Function`. If you split findSerializerByName it will be ok. Also I think putting getTypes in `SerializationSymbols.ts` does not makes sense. It should be in SerializationService as a private method. 

```suggestion
        const keyForClass = getKeyForClass(clazzType);
```